### DISCUSSION TITLE
gatsby-source-wordpress: WP categories return undefined

### DISCUSSION THREAD

**Discussion by opr-inzn** [gatsbyjs/gatsby#34944] [Help] [answered]
### Description

There seems to be an error in the Gatsby WPGraphQL Plugin. In GraphQL under localhost:8000/___graphql I get my categories name and id returned for a post with the following query:

```
query MyQuery {
  wpPost(title: { regex: "/Works/" }) {
    categories {
      nodes {
        id
      }
    }
  }
}
```
<img width="840" alt="Screenshot 2022-02-27 at 12 19 40 PM" src="https://user-images.githubusercontent.com/80823111/155880448-8261ff24-974c-4707-bd4a-1ead563e5c27.png">

However, in Gatsby the same query does not return anything, when I say

`<p>{data.wpPost.categories.nodes.id}</p>`

there is no error and the paragraph element get's rendered but without any content. I have also tried it with name, slug and basically any other of the available elements in "category" with the same result: an empty element.

Any help would be greatly appreciated!

### Expected Result

`<p>dGVybTo3OQ==</p>`

### Actual Result

`<p></p>`

### Environment

```shell
System:
    OS: macOS 12.2.1
    CPU: (10) arm64 Apple M1 Pro
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.5.0 - /opt/homebrew/bin/node
    npm: 8.4.1 - /opt/homebrew/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 98.0.4758.109
    Safari: 15.3
  npmPackages:
    gatsby: ^4.0.0 => 4.8.0
    gatsby-plugin-image: ^2.0.0 => 2.8.0
    gatsby-plugin-manifest: ^4.0.0 => 4.8.0
    gatsby-plugin-offline: ^5.0.0 => 5.8.0
    gatsby-plugin-react-helmet: ^5.0.0 => 5.8.0
    gatsby-plugin-sharp: ^4.0.0 => 4.8.0
    gatsby-source-filesystem: ^4.0.0 => 4.8.0
    gatsby-source-wordpress: ^6.0.0 => 6.8.0
    gatsby-transformer-sharp: ^4.0.0 => 4.8.0
  npmGlobalPackages:
    gatsby-cli: 4.8.0
```

**Comment by ekon97** [comment, marked as answer]
Could you try data.wpPost.categories.nodes[0].id ?

**Comment by opr-inzn** [comment]
I was totally stupid, that fixed it! Thanks a lot!!
